### PR TITLE
OSDOCS-8427: Dashboard enhancements for lokiless use

### PR DESCRIPTION
**Pull request by skrthomas**
For peer review/merge review:
Merge to only the no-1.5 branch - no cherrypicks are required.
This PR is part of an experiment for simplifying merges for asynchronous content, and I will open one PR against main to incorporate all of the Network Observability 1.5 content just before its GA.

Version(s):

4.11 +
Issue:

https://issues.redhat.com/browse/OSDOCS-8427
Link to docs preview:

This [Overview of Network Observability metrics](https://docs.openshift.com/container-platform/4.14/network_observability/network-observability-overview.html#network-observability-dashboards) is being replaced by the below dedicated Metrics page:
- Viewing Network Observability metrics dashboards: https://69106--docspreview.netlify.app/openshift-enterprise/latest/network_observability/metrics-alerts-dashboards#network-observability-viewing-metrics-dashboards_metrics-dashboards-alerts
- Network Observability metrics: https://69106--docspreview.netlify.app/openshift-enterprise/latest/network_observability/metrics-alerts-dashboards#network-observability-metricsmetrics-dashboards-alerts
- Creating alerts: https://69106--docspreview.netlify.app/openshift-enterprise/latest/network_observability/metrics-alerts-dashboards#network-observability-netobserv-dashboard-high-traffic-alert_metrics-dashboards-alerts
- Network Observability Loki rate limit alerts: https://69106--docspreview.netlify.app/openshift-enterprise/latest/network_observability/network-observability-operator-monitoring#network-observability-netobserv-dashboard-rate-limit-alerts_network_observability (Because of a peer review suggestion to make includeList alert example into a full procedure, rather than linking to the Monitoring documentation for procedural information, I decided to change the Loki rate limit alert module as well so all the alerting modules are consistent.)

QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/network-observability-metrics.adoc
Comment (jpinsonneau) [inline modules/network-observability-metrics.adoc:39]:
```suggestion

When the `DNSTracking` feature is enabled in `spec.agent.ebpf.features`, the following additional metrics are available:

* `namespace_dns_latency_seconds` *

```

File: network_observability/metrics-alerts-dashboards.adoc
Comment (skrthomas) [inline network_observability/metrics-alerts-dashboards.adoc:13]: @jpinsonneau @jotak I think we can put the metrics alerting examples here? Just let me know what you'd like me to show here in terms of includeList examples

Comment (jpinsonneau) [reply network_observability/metrics-alerts-dashboards.adoc:13]: I've just noticed custom metrics PR is still WIP:
https://github.com/netobserv/network-observability-operator/pull/445/files#diff-f56c4ef1268ddd69d4da318cd3ffc855de718f70c7de5de760fd18ef69f79aeb

@jotak do we still plan to have it as dev preview for 1.5 ?

Comment (skrthomas) [reply network_observability/metrics-alerts-dashboards.adoc:13]: I guess, I was thinking the example here would be includeList metrics and alerts examples. In 1.6 timeframe, the custom metrics can live here as well.

File: modules/network-observability-metrics.adoc
Comment (jotak) [inline modules/network-observability-metrics.adoc:8]: Maybe mention that when looking for these metrics in Prometheus (e.g. in the Console through menu Observe > Metrics), all these names are prefixed "netobserv_", such as "netobserv_namespace_flows_total"

File: network_observability/metrics-alerts-dashboards.adoc
Comment (jotak) [reply network_observability/metrics-alerts-dashboards.adoc:13]: @skrthomas for an alerting example I'm suggesting one here: https://github.com/netobserv/documents/pull/62
Yaml is:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: netobserv-alerts
spec:
  groups:
  - name: NetObservAlerts
    rules:
    - alert: NetObservIncomingBandwidth
      annotations:
        message: |-
          {{ $labels.job }}: incoming traffic exceeding 10 MBps for 30s on {{ $labels.DstK8S_OwnerType }} {{ $labels.DstK8S_OwnerName }} ({{ $labels.DstK8S_Namespace }}).
        summary: "High incoming traffic."
      expr: sum(rate(netobserv_workload_ingress_bytes_total{SrcK8S_Namespace="openshift-ingress"}[1m])) by (job, DstK8S_Namespace, DstK8S_OwnerName, DstK8S_OwnerType) > 10000000
      for: 30s
      labels:
        severity: warning
```

This is an alert when incoming traffic reaches a given threshold (here 10 MBps) per destination workload

Comment (skrthomas) [reply network_observability/metrics-alerts-dashboards.adoc:13]: Thank you so much :)

Comment (skrthomas) [reply network_observability/metrics-alerts-dashboards.adoc:13]: @jotak would you happen to have an example for includeList? 😇

Comment (skrthomas) [reply network_observability/metrics-alerts-dashboards.adoc:13]: I see the includeList metric, `netobserv_workload_ingress_bytes_total` is in the expression. In the includeList reference, the intro text states: 
> ...configurable in the `spec.processor.metrics.includeList` of the `FlowCollector` custom resource to add or remove metrics.

So, like your comment there says, I think I need to add more context about when a user creates an alert, they specify the includeList metric in the Prometheus rule.

File: modules/network-observability-metrics.adoc
Comment (skrthomas) [reply modules/network-observability-metrics.adoc]: Okay, I adjusted the wording a bit. I also feel like its insufficient to just mention configuration in the `FlowCollector` when you can also specify these metrics in a Prometheus rule, so I have both mentioned now.

File: network_observability/metrics-alerts-dashboards.adoc
Comment (jotak) [reply network_observability/metrics-alerts-dashboards.adoc:14]: This metric I used is present by default so users who didn't configure `includeList` explicitly wouldn't have to worry about it. But I agree, we could say something like: "Make sure the metric `spec.processor.metrics.includeList` either includes `workload_ingress_bytes_total`, or is unset since this metric is enabled by default."

Comment (skrthomas) [reply network_observability/metrics-alerts-dashboards.adoc:14]: I added this as a callout in the YAML: https://github.com/openshift/openshift-docs/pull/69106/files#diff-8d0022591f4f4567d0cc2ff90110d33839696453336a9fd67a8d766ff79f3f48R31

File: modules/network-observability-metrics.adoc
Comment (Amoghrd) [inline modules/network-observability-metrics.adoc:12]: `Names followed by `*` are enabled by default:` line can be removed here. since its mentioned in L16 below

File: modules/network-observability-metrics.adoc
Comment (Amoghrd) [inline modules/network-observability-metrics.adoc:46]: There are some more DNS metrics which are not listed here. You can find the full list of metrics here https://github.com/netobserv/network-observability-operator/blob/main/docs/Metrics.md

File: modules/network-observability-metrics.adoc
Comment (Amoghrd) [inline modules/network-observability-metrics.adoc:48]: This is `FlowRTT` here

File: modules/network-observability-metrics.adoc
Comment (skrthomas) [reply modules/network-observability-metrics.adoc:46]: Ah man, idk how I missed these. Thanks for pointing them out!

File: modules/network-observability-includelist-example.adoc
Comment (jotak) [inline modules/network-observability-includelist-example.adoc:6]: Sorry for the late suggestion, but when reading this we could think we can just create alerts for that very precise use case.

It could perhaps be reformulated as "Creating alerts" as a title, change the first paragraph sentence to something like:

> You can create custom rules for the *Netobserv* metrics to trigger alerts when some defined conditions are met.

And then finally introduce that special case on ingress traffic like you did.

wdyt?

Comment (skrthomas) [reply modules/network-observability-includelist-example.adoc:6]: No worries, any and all feedback at any time is good :)  I can see what you mean here. This procedure could be misinterpreted as the _only_ for this use case. I take your suggestion

Comment (skrthomas) [reply modules/network-observability-includelist-example.adoc]: Addressed. TY!

**Review by jotak — APPROVED** [review]

File: modules/network-observability-metrics.adoc
Comment (bergerhoffer) [inline modules/network-observability-metrics.adoc:12]: Markup suggestions

```suggestion
When looking for these metrics in Prometheus, such as in the Console through *Observe* -> *Metrics*, or when defining alerts, all the metrics names are prefixed with `netobserv_`. For example, `netobserv_namespace_flows_total`. Available metrics names are as follows.
```

File: modules/network-observability-metrics.adoc
Comment (bergerhoffer) [inline modules/network-observability-metrics.adoc:16]:
```suggestion
Names followed by an asterisk (`*`) are enabled by default. 
```

File: modules/network-observability-metrics.adoc
Comment (bergerhoffer) [inline modules/network-observability-metrics.adoc:27]: This list isn't quite in alphabetical order in a few places, just checking if it should be

File: modules/network-observability-metrics.adoc
Comment (bergerhoffer) [inline modules/network-observability-metrics.adoc:15]: Just checking - does this do anything?

File: modules/network-observability-metrics.adoc
Comment (bergerhoffer) [inline modules/network-observability-metrics.adoc:50]: Depending on whether you change your main heading for these lists, just thinking whether it might be nice to have each of these with a more visible title. Just something to consider, totally fine to leave it as is too!

File: modules/network-observability-viewing-dashboards.adoc
Comment (bergerhoffer) [inline modules/network-observability-viewing-dashboards.adoc:8]: Consider 
Suggestion for active voice

```suggestion
On the *Overview* tab in the {product-title} console, you can view the overall aggregated metrics of the network traffic flow on the cluster. You can choose to display the information by node, namespace, owner, pod, and service. You can also use filters and other display options to further refine the metrics.
```

File: modules/network-observability-viewing-dashboards.adoc
Comment (bergerhoffer) [inline modules/network-observability-viewing-dashboards.adoc:9]: Missing the procedure heading

```suggestion
.Procedure
```

File: modules/network-observability-viewing-dashboards.adoc
Comment (bergerhoffer) [inline modules/network-observability-viewing-dashboards.adoc:10]: Might be nice to clarify that it's in the web console.

File: modules/network-observability-viewing-dashboards.adoc
Comment (bergerhoffer) [inline modules/network-observability-viewing-dashboards.adoc:11]: Not sure if there's a reason it's written like this, or if it can lose the second "and"

```suggestion
. View distillations of the network traffic metrics in the following categories, with each having the subset per node, namespace, source, and destination:
```

File: modules/network-observability-viewing-dashboards.adoc
Comment (bergerhoffer) [inline modules/network-observability-viewing-dashboards.adoc:18]: Just checking if it's necessary to state this, if they're still in the same page

```suggestion
Select the *Netobserv/Health* dashboard. 
```

File: modules/network-observability-viewing-dashboards.adoc
Comment (bergerhoffer) [inline modules/network-observability-viewing-dashboards.adoc:19]:
```suggestion
. View metrics about the health of the Operator in the following categories, with each having the subset per node, namespace, source, and destination.
```

File: network_observability/metrics-alerts-dashboards.adoc
Comment (bergerhoffer) [inline network_observability/metrics-alerts-dashboards.adoc:9]: * If it maintains the same correct meaning, one word instead of two
* "of" is missing with the "make use of". Or there's probably a better singular word that could work too. I just can't think of a good one to suggest at the moment :) Maybe "utilize", though I know a lot of times people say "utilize" you can just mean "use". But maybe it would be appropriate here :shrug: 

```suggestion
The Network Observability Operator uses the `flowlogs-pipeline` to generate metrics from flow logs. You can utilize these metrics by setting custom alerts and viewing dashboards.
```

File: network_observability/metrics-alerts-dashboards.adoc
Comment (bergerhoffer) [inline network_observability/metrics-alerts-dashboards.adoc:17]: Normally I think we want these additional resources to just be single links, without text around them.

Also, should there be an xref either here or somewhere else in the assembly for the "Creating an alert for high ingress traffic" you mention earlier?

File: network_observability/network-observability-operator-release-notes.adoc
Comment (bergerhoffer) [inline network_observability/network-observability-operator-release-notes.adoc:67]: The title of the page this links to is called "Viewing Network Observability metrics dashboards". Suggest matching that here

File: network_observability/network-observability-operator-release-notes.adoc
Comment (bergerhoffer) [inline network_observability/network-observability-operator-release-notes.adoc:148]: Same comment about matching the title here

File: network_observability/network-observability-overview.adoc
Comment (bergerhoffer) [inline network_observability/network-observability-overview.adoc:26]: Just checking if this "overview" mention here needs to be adjusted with this section below being removed?

File: network_observability/observing-network-traffic.adoc
Comment (bergerhoffer) [inline network_observability/observing-network-traffic.adoc:19]: Same comment about preferring these to be single xrefs. If you do keep it as a sentence, the below one is missing a period

File: network_observability/observing-network-traffic.adoc
Comment (bergerhoffer) [inline network_observability/observing-network-traffic.adoc:21]: It looks like this was meant to be replaced in the link text above instead of floating here

File: network_observability/observing-network-traffic.adoc
Comment (bergerhoffer) [inline network_observability/observing-network-traffic.adoc:28]: Same comment about the xrefs, and missing a period if you keep it a sentence

**Review by bergerhoffer — COMMENTED** [review]
Very cool to be experimenting with using a branch other than main to coordinate asynchronous content. Do we have any guidelines on this now (like getting approval before doing so, how to coordinate the merge when it's time - like making sure no other unintended changes came into it before it's merged to main)? Or will we plan on making some guidelines if this experiment is successful? I'll be curious to hear how it goes!

Left some comments, but overall nice job!

File: modules/network-observability-includelist-example.adoc
Comment (skrthomas) [reply modules/network-observability-includelist-example.adoc:7]: i think it does, and we bold it elsewhere, because its the name of the dashboard in Observe -> Dashboards in the OCP web console GUI.

File: _topic_maps/_topic_map.yml
Comment (skrthomas) [reply _topic_maps/_topic_map.yml:2654]: I know :( I was really wanting to use commas, but I can't use commas in the topic_map.yml so I made it inconsistent. However, maybe I should just title the assembly the same. Good question.

Comment (skrthomas) [reply _topic_maps/_topic_map.yml:2654]: Addressed.

File: modules/network-observability-includelist-example.adoc
Comment (skrthomas) [reply modules/network-observability-includelist-example.adoc:31]: Yeah looks like I am missing the netobserv prefix here.

Comment (skrthomas) [reply modules/network-observability-includelist-example.adoc:31]: Addressed!

File: modules/network-observability-metrics.adoc
Comment (skrthomas) [reply modules/network-observability-metrics.adoc:15]: I don't think this [options="header"] actually does anything. I was thinking I needed it but I don't know why I was thinking that because I can't find anything in asciidoc writing reference or contributing guide that says we need this kind of thing. I've removed it to check and make sure.

File: modules/network-observability-metrics.adoc
Comment (skrthomas) [reply modules/network-observability-metrics.adoc:50]: This is a good suggestion. I will rework these headings for visibility!

File: modules/network-observability-viewing-dashboards.adoc
Comment (skrthomas) [reply modules/network-observability-viewing-dashboards.adoc:9]: 🤦‍♀️

File: modules/network-observability-viewing-dashboards.adoc
Comment (skrthomas) [reply modules/network-observability-viewing-dashboards.adoc:11]: I think in conversation, source and destination get lumped together as hand-in-hand but I agree they don't necessarily need to be written like this.

File: modules/network-observability-viewing-dashboards.adoc
Comment (skrthomas) [reply modules/network-observability-viewing-dashboards.adoc:18]: Good pt.

File: network_observability/metrics-alerts-dashboards.adoc
Comment (skrthomas) [reply network_observability/metrics-alerts-dashboards.adoc:17]: Yeah that is right. I will remove the surrounding text. 

The include for Creating an alert for high-ingress traffic is this: network-observability-includelist-example.adoc. The title of it has changed though to, just Creating alerts so I will adjust that reference in the module, network-observability-metrics.adoc

File: modules/network-observability-includelist-example.adoc
Comment (skrthomas) [reply modules/network-observability-includelist-example.adoc]: I agree. I think I made this a concept because I am sorta punting to the linked Monitoring doc for the true procedural info, ofc its linked in the Assembly. I always get torn with NetObserv docs about how much to replicate from Logging/Monitoring docs because both are entwined with Network Observability in use cases, but I can also see how its just helpful to have those steps flat out for the customer, and still keep the link if they want more general info about creating Prometheus alerts.

File: network_observability/network-observability-overview.adoc
Comment (skrthomas) [reply network_observability/network-observability-overview.adoc]: yeah I think initially I removed this Overview mention totally in favor of a dedicated assembly, but I actually think we should mention it briefly and link to the new assemblies. Thanks for pointing this out!

File: modules/network-observability-includelist-example.adoc
Comment (skrthomas) [inline modules/network-observability-includelist-example.adoc:16]: @jotak I just want to confirm if its OK to present the example like this? Creating the YAML file in the web console, and the prerequisites are OK?

Comment (jotak) [reply modules/network-observability-includelist-example.adoc:16]: nice, everything looks good!

File: modules/network-observability-viewing-alerts.adoc
Comment (mburke5678) [inline modules/network-observability-viewing-alerts.adoc:34]:
```suggestion
. View the metrics about the health of the Operator that are displayed on the page.
```

File: modules/network-observability-viewing-dashboards.adoc
Comment (mburke5678) [inline modules/network-observability-viewing-dashboards.adoc:12]: Is _distillations_ the best word? I am concerned about translation.

File: network_observability/observing-network-traffic.adoc
Comment (mburke5678) [inline network_observability/observing-network-traffic.adoc:20]: This link text should probably be Network Observability metrics, as in [below](https://github.com/openshift/openshift-docs/pull/69106/files#diff-21cba4b9eea6286345ee8e2ad7b3cc0e4dc6ac125a47623a7df9f52f2f2506aaR27).

```suggestion
* xref:../network_observability/metrics-alerts-dashboards.adoc#network-observability-metrics_metrics-dashboards-alerts[Network Observability metrics]
```

**Comment by mburke5678** [comment]
@skrthomas I found a few nits that I am hoping you can look at before we merge.

File: modules/network-observability-rate-limit-alert.adoc
Comment (mburke5678) [inline modules/network-observability-rate-limit-alert.adoc:1]: // Module included in the following assemblies:

Comment (mburke5678) [reply modules/network-observability-rate-limit-alert.adoc:1]: Is this module included in any assemblies? I'm not seeing an `include;;` for it

File: modules/network-observability-rate-limit-alert.adoc
Comment (mburke5678) [inline modules/network-observability-rate-limit-alert.adoc:15]: We should add the steps to get to this point in the console.

File: modules/network-observability-rate-limit-alert.adoc
Comment (skrthomas) [reply modules/network-observability-rate-limit-alert.adoc]: Added, the include comment was wrong.

**Pull request merged**
